docs/core/diagnostics/faq-dumps.yml

```diff
@@ -11,15 +11,48 @@ sections:
+
+          If the calls aren't intercepted then the kernel will do a variety of built-in access checks. The docs for [ptrace()](https://www.man7.org/linux/man-pages/man2/ptrace.2.html) include a detailed description near the bottom titled "Ptrace access mode checking" that describes how these are done. Accessing the /proc filesystem also uses a variation of the same ptrace access mode checking. This is an abbreviated summary of the security checks performed and some places we have discovered where access might be denied:
+
+          1. Either the calling process needs to have the same user id as the target process, or the calling process needs to have CAP_SYS_PTRACE. If neither of these is true access is denied. Since the .NET runtime doesn't do anything to change the user account when launching createdump the user ids should match and this step should succeed.
+          2. If createdump doesn't have CAP_SYS_PTRACE (it doesn't by default), then the target process being dumped needs to be marked "dumpable". By default most processes on Linux are dumpable but this setting can be changed by calling [prctl()](https://www.man7.org/linux/man-pages/man2/prctl.2.html) with the PR_SET_DUMPABLE option. If you add capabilities to a process using the setcap tool, this can also cause a process to stop being dumpable. See [the Linux documentation](https://www.man7.org/linux/man-pages/man2/PR_SET_DUMPABLE.2const.html) for a more detailed description of the dumpable setting and what causes it to be disabled.
+          3. All of the enabled [Linux security modules](https://www.kernel.org/doc/html/v4.16/admin-guide/LSM/index.html) (LSMs) are enumerated and each of them must approve the access. Unforetunately if an LSM denies the access there is no uniform Linux reporting mechanism to know which one is responsible. Instead you need to determine which ones are enabled on your system and then investigate each individually. You can determine which LSMs are active by running: `cat /sys/kernel/security/lsm`. Although any LSM could be responsible, [Yama](https://www.kernel.org/doc/html/v4.16/admin-guide/LSM/Yama.html), [SELinux](https://selinuxproject.org/page/Main_Page), and [AppArmor](https://gitlab.com/apparmor/apparmor/-/wikis/GettingStarted) are frequently the relevant ones.
```

Comment:
```suggestion
          - All of the enabled [Linux security modules](https://www.kernel.org/doc/html/v4.16/admin-guide/LSM/index.html) (LSMs) are enumerated and each of them must approve the access. Unfortunately, if an LSM denies the access, there's no uniform Linux reporting mechanism to know which one is responsible. Instead you need to determine which ones are enabled on your system and then investigate each individually. You can determine which LSMs are active by running: `cat /sys/kernel/security/lsm`. Although any LSM could be responsible, [Yama](https://www.kernel.org/doc/html/v4.16/admin-guide/LSM/Yama.html), [SELinux](https://selinuxproject.org/page/Main_Page), and [AppArmor](https://gitlab.com/apparmor/apparmor/-/wikis/GettingStarted) are frequently the relevant ones.
```

Reply:
Unless they are ordered steps, they should be bullets, not a numbered list.

Reply:
They are ordered in the sense that the kernel code will execute the checks in the sequence written. I'm not sure if the ordering matters all that much though. Either they all succeed or they don't.